LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDevice.cs

```diff
@@ -421,6 +431,78 @@ internal async Task<bool> UpdateAfterJoinAsync(string devAddr, string nwkSKey, s
             return succeeded;
         }
 
+        internal void SetRequestHandler(ILoRaDataRequestHandler dataRequestHandler) => this.dataRequestHandler = dataRequestHandler;
+
+        public void Queue(LoRaRequest request)
+        {
+            // Access to runningRequest and queuedRequests must be
+            // thread safe
+            lock (this)
+            {
+                if (this.runningRequest == null)
+                {
+                    this.StartNextRequest(request);
+                }
+                else
+                {
+                    this.queuedRequests.Enqueue(request);
+                }
+            }
+        }
+
+        private void QueueNext()
```

Comment:
The name is confusing to me. You are not queueing anything, but dequeuing and processing. Maybe ProcessNext would be an option?